server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -205,6 +250,103 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
         }
     }
 
+    private long getPoolInterval(Product prod) {
+        long interval = maxDevLifeDays;
+        String prodThenString = prod.getAttributeValue("expired_after");
+        if (prodThenString != null && Long.parseLong(prodThenString) < maxDevLifeDays) {
+            interval = Long.parseLong(prodThenString);
+        }
+        return 1000 * 60 * 60 * 24 * interval;
+    }
+
+    private Pool getDevPool(Consumer consumer, String sku) {
+        PoolFilterBuilder poolFilters = new PoolFilterBuilder();
+        poolFilters.addAttributeFilter(Pool.REQUIRES_CONSUMER_ATTRIBUTE, consumer.getUuid());
+        poolFilters.addAttributeFilter(Pool.DEVELOPMENT_POOL_ATTRIBUTE, "true");
+        Page<List<Pool>> poolsPage = poolManager.listAvailableEntitlementPools(consumer, null,
+                consumer.getOwner(), null, null, true, true, poolFilters, null);
+        if (poolsPage != null &&
+            poolsPage.getPageData() != null &&
+            poolsPage.getPageData().size() == 1) {
+            return poolsPage.getPageData().get(0);
+        }
+        else {
+            return null;
+        }
+    }
+
+    protected Pool assembleDevPool(Consumer consumer, String sku) {
+        Set<Product> providedProducts = new HashSet<Product>();
+        Date now = new Date();
+        List<String> devProductIds = new ArrayList<String>();
+        Product skuProd = null;
+
+        devProductIds.add(sku);
+        for (ConsumerInstalledProduct ip : consumer.getInstalledProducts()) {
+            devProductIds.add(ip.getProductId());
+        }
+        List<Product> prods = productAdapter.getProductsByIds(consumer.getOwner(), devProductIds);
+        if (prods.size() != devProductIds.size()) {
+            // determine exception message
+            produceMissingDevProductException(devProductIds, prods);
+        }
```

Comment:
I think it would be better to rename the produceMissingDevProductException method to verifyDevProducts and move the 'if (prods.size() != devProductIds.size())' check inside it.